app/models/transformation_mapping.rb

```diff
@@ -1,9 +1,53 @@
 class TransformationMapping < ApplicationRecord
+  VM_VALID = N_("OK").freeze
+
   has_many :transformation_mapping_items, :dependent => :destroy
 
   validates :name, :presence => true, :uniqueness => true
 
   def destination(source)
     transformation_mapping_items.find_by(:source => source).try(:destination)
   end
+
+  def select_vms
+    valid_list = []
+
+    Vm.each do |vm|
```

Comment:
While the `.each` method does not exist on the `Vm` model is this really what we want to do?  It will return every infra and cloud VM object in the database.  It should at least be limited to the source provider defined in the mapping.

Reply:
You can aggregate all source clusters from the mapping and find all VMs under the clusters.